File: src/main/java/codereview/simpleorder/item/consumer/ItemEventListener.java
package codereview.simpleorder.item.consumer;

import codereview.simpleorder.item.application.ItemService;
import codereview.simpleorder.item.dto.event.OrderCreatedEvent;
import lombok.RequiredArgsConstructor;
import org.springframework.context.event.EventListener;
import org.springframework.stereotype.Component;
import org.springframework.transaction.event.TransactionalEventListener;

import java.util.Map;

@Component
@RequiredArgsConstructor
public class ItemEventListener {

    private final ItemService itemService;

    @TransactionalEventListener
    public void captureEvent(OrderCreatedEvent event) {

        Map<Long, Integer> itemIdToDecreaseQuantity = event.values();
        itemService.decreaseItems(itemIdToDecreaseQuantity);
    }
}

File: item/src/main/kotlin/msa/with/ddd/item/consumer/ItemEventListener.kt
package msa.with.ddd.item.consumer

import msa.with.ddd.item.application.ItemService
import msa.with.ddd.item.dto.order.OrderCreatedEvent
import org.springframework.context.event.EventListener
import org.springframework.stereotype.Component

@Component
class ItemEventListener(private val itemService: ItemService) {

    @EventListener
    fun captureEvent(event: OrderCreatedEvent) {
        val itemIdToDecreaseQuantity = event.values()
        itemService.decreaseItems(itemIdToDecreaseQuantity)
    }
}